api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -165,15 +165,21 @@ message Bootstrap {
   // Configuration for internal processing of stats.
   metrics.v3.StatsConfig stats_config = 13;
 
-  // Optional duration between flushes to configured stats sinks. For
-  // performance reasons Envoy latches counters and only flushes counters and
-  // gauges at a periodic interval. If not specified the default is 5000ms (5
-  // seconds).
-  // Duration must be at least 1ms and at most 5 min.
-  google.protobuf.Duration stats_flush_interval = 7 [(validate.rules).duration = {
-    lt {seconds: 300}
-    gte {nanos: 1000000}
-  }];
+  oneof stats_flush {
+    // Optional duration between flushes to configured stats sinks. For
+    // performance reasons Envoy latches counters and only flushes counters and
+    // gauges at a periodic interval. If not specified the default is 5000ms (5
+    // seconds).
+    // Duration must be at least 1ms and at most 5 min.
+    google.protobuf.Duration stats_flush_interval = 7 [(validate.rules).duration = {
+      lt {seconds: 300}
+      gte {nanos: 1000000}
+    }];
+
+    // Flush stats to sinks only when queried for on the admin interface. Stats flushing
+    // defaults to the timer
+    bool stats_flush_on_admin = 29 [(validate.rules).bool = {const: true}];
```

Comment:
Not an ideal solution, but using this field can supersede the interval usage.
Another option is to use both interval and admin flush concurrently.

Reply:
How would they be used concurrently? By setting the `stats_flush_on_admin`, we disable the timer. If they can be set concurrently, is there a way we can indicate to not flush on a timer? Maybe a duration of 0s?

Reply:
I think the current suggestion makes sense in terms of API compliance.
Can you please also update `envoy/docs/root/operations/admin.rst`?

Reply:
lmk how it looks

Reply:
Thanks!

Reply:
Doing final API review here, @hamdiallam can you comment on the use case here? I'm trying to figure out if we ever would want an option for flushing with timer _and_ with admin endpoint access?

Reply:
We don't ever want an option for flushing with the timer and admin endpoint access. For backwards compatibility, we can't use `oneof` to enforce that outside of adding the `oneof_promotion` annotation which in the v4 api, becomes a `oneof`. 

If both are specified in v3, admin endpoint access takes precedence. Perhaps a warning log entry that both were specified?

Reply:
> If both are specified in v3, admin endpoint access takes precedence. Perhaps a warning log entry that both were specified?

I would make this a config failure similar to what would be done if it were a oneof. This is what we do elsewhere.

Reply:
Can you update the docs for this field as well as the other one to say that only one of them can be set?